### PR TITLE
Rename server functions to follow *_loop convention

### PR DESCRIPTION
**Pull request by kachayev**
Follow up on https://github.com/async-rs/a-chat/pull/7.

Covers https://github.com/async-rs/async-std/issues/106.

Hope I didn't miss anything.

### CODE REVIEW

**Comment by matklad** [comment]
changes look good to me, @kachayev could you resolve the merge conflcts?

**Comment by yoshuawuyts** [comment]
@kachayev hi, just checking in on this patch. This seems like something we'd definitely want to merge; do you think you can get around to rebasing? We'd appreciate it a lot!

**Comment by kachayev** [comment]
@yoshuawuyts Oh, sorry for the delay 🤦‍♂ Will do it shortly!

**Comment by kachayev** [comment]
@yoshuawuyts Seems good now!

File: docs/src/tutorial/accept_loop.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>
 
 Now we can write the server's accept loop:
 
-```rust,edition2018
```
Comment (yoshuawuyts) [inline docs/src/tutorial/accept_loop.md]: It seems `edition2018` should be kept, as it's causing the tests to fail https://travis-ci.com/async-rs/async-std/jobs/237051125

Comment (kachayev) [reply docs/src/tutorial/accept_loop.md]: Yep, my bad 👍

**Comment by yoshuawuyts** [comment]
Looks like there's still come compilation errors: https://travis-ci.com/async-rs/async-std/jobs/237256929

**Comment by kachayev** [comment]
@yoshuawuyts  Compiles now ;)

**Comment by matklad** [comment]
Great, thanks for following up here @kachayev !

**Pull request merged**